lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -1127,13 +1164,25 @@ SEARCHFORM
         ),
         @foreign_cols,    # already assembled from quoted identifiers
         @custom_cols,
+        @join_cols
     );
     my $add_actions
         = $args->{editable}
         ? ", $table_name.$key_column AS actions"
         : '';
     my $query = "SELECT $col_list $add_actions FROM $table_name";
+    if (my $joins = $args->{joins}) {
+        for my $joiner (@$joins) {
+            my $join_style = $joiner->{join_style} || "join";
+            my $join_table = $joiner->{db_table} || die "'joins' directive needs 'db_table' setting";
+            my $join_on = $joiner->{join_on} || die "'joins' directive needs 'join_on' setting";
+            my @join_on_eq = %$join_on; # key and value flattened
+            $query .= " $join_style $join_table on $join_on_eq[0]=$join_on_eq[1]";
```

Comment:
Could this part be handled by passing the conditional hashref to the existing where-clause generation, to avoid SQL injection problems and allow more flexibility?